vpd-tool/include/tool_utils.hpp

```diff
@@ -125,5 +126,106 @@ inline std::string getPrintableValue(const types::BinaryVector& i_keywordValue)
 
     return l_oss.str();
 }
+
+/**
+ * @brief API to save keyword's value on file.
+ *
+ * API writes keyword's value on the given file path. If the data is in hex
+ * format, API strips '0x' and saves the value on the given file.
+ *
+ * @param[in] i_filePath - File path.
+ * @param[in] i_keywordValue - Keyword's value.
+ *
+ * @return - true on successfully writing to file, false otherwise.
+ */
+inline bool saveToFile(const std::string& i_filePath,
+                       const std::string& i_keywordValue)
+{
+    bool l_returnStatus = false;
+
+    if (i_keywordValue.empty())
+    {
+        // ToDo: log only when verbose is enabled
+        std::cerr << "Save to file[ " << i_filePath
+                  << "] failed, reason: Empty keyword's value received"
+                  << std::endl;
+        return l_returnStatus;
+    }
+
+    std::error_code l_ec;
+    auto l_baseFolder = std::filesystem::path(i_filePath).parent_path();
+
+    if (!l_baseFolder.empty() && !std::filesystem::exists(l_baseFolder, l_ec))
```

Comment:
added this check, because we wont get any exception if  base folder doesn't exists & is_open() will also return true if  folder doesn't exists.